werkzeug/wrappers.py

```diff
@@ -669,12 +669,20 @@ def remote_addr(self):
 
         .. versionadded:: 0.7''')
 
-    is_xhr = property(lambda x: x.environ.get('HTTP_X_REQUESTED_WITH', '')
-                      .lower() == 'xmlhttprequest', doc='''
-        True if the request was triggered via a JavaScript XMLHttpRequest.
+    @property
+    def is_xhr(self):
+        """True if the request was triggered via a JavaScript XMLHttpRequest.
         This only works with libraries that support the `X-Requested-With`
         header and set it to "XMLHttpRequest".  Libraries that do that are
-        prototype, jQuery and Mochikit and probably some more.''')
+        prototype, jQuery and Mochikit and probably some more. This property
+        is deprecated."""
```

Comment:
http://www.sphinx-doc.org/en/stable/markup/para.html#directive-deprecated

~~~rst
.. deprecated:: 2.10
   ``X-Requested-With`` is not standard and is unreliable.
~~~